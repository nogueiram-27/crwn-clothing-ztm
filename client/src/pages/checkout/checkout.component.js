import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect'
import { selectCartItems , selectCartTotal } from '../../redux/cart/cart.selectors'

import CheckoutItem from '../../components/checkout-item/checkout-item.component'
import StripeCheckoutButton from '../../components/stripe-button/stripe-button.component';

import { CheckoutContainer, HeaderContainer, HeaderBlockContainer, TotalContainer, TestWarningContainer } from './checkout.styles'

const CheckoutPage = ({ cartItems , cartTotal }) => (
    <CheckoutContainer>
        <HeaderContainer>
            <HeaderBlockContainer>
                <span>Product</span>
            </HeaderBlockContainer>
            <HeaderBlockContainer>
                <span>Description</span>
            </HeaderBlockContainer>
            <HeaderBlockContainer>
                <span>Quantity</span>
            </HeaderBlockContainer>
            <HeaderBlockContainer>
                <span>Price</span>
            </HeaderBlockContainer>
            <HeaderBlockContainer>
                <span>Remove</span>
            </HeaderBlockContainer>
        </HeaderContainer>
        {
            cartItems.map(cartItem => 
                    <CheckoutItem key={cartItem.id} cartItem={cartItem} />
                )
        }
        <TotalContainer>
            <span>TOTAL:${cartTotal}</span>
        </TotalContainer>
        <TestWarningContainer>
            *Please use the following test credit card for payments*
            <br />
            4242 4242 4242 4242 - Exp:01/23 - CVV: 123
        </TestWarningContainer>
        <StripeCheckoutButton price={cartTotal} />
    </CheckoutContainer>
)

const mapStateToProps = createStructuredSelector ({
    cartItems: selectCartItems,
    cartTotal: selectCartTotal
})

export default connect(mapStateToProps)(CheckoutPage)