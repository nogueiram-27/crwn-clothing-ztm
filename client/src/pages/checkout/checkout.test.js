import React from 'react'
import { shallow } from 'enzyme'

import { CheckoutPage } from './checkout.component'
import CheckoutItem from '../../components/checkout-item/checkout-item.component'

describe('CheckoutPage', () => {
    let wrapper;
    let mockCartItems = [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }]

    beforeEach(() => {
        const mockProps = {
            cartItems: mockCartItems,
            cartTotal: 100
        }

        wrapper = shallow(<CheckoutPage {...mockProps } />)
    })

    it('should render CheckoutPage component', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('should render CheckoutItem component as many times as mockItems size', () => {
        expect(wrapper.find(CheckoutItem).length).toBe(mockCartItems.length)
    })
})
