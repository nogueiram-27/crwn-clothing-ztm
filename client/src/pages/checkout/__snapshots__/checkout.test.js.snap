// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutPage should render CheckoutPage component 1`] = `ShallowWrapper {}`;
